### PR TITLE
ARROW-12833: [JS] Construct perf data in JS

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: js/perf/config.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import * as Arrow from '../src/Arrow.dom';
+
+console.time('Prepare Data');
+
+const LENGTH = 100000;
+const NUM_BATCHES = 10;
+
+const values = Arrow.Utf8Vector.from(['Charlottesville', 'New York', 'San Francisco', 'Seattle', 'Terre Haute', 'Washington, DC']);
+
```
Comment (trxcllnt) [inline js/perf/config.ts:26]:
```suggestion
const originType = new Arrow.Dictionary(values.type, new Arrow.Int8, 0, false);
const destinationType = new Arrow.Dictionary(values.type, new Arrow.Int8, 0, false);

```

File: js/perf/config.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        () => (Math.random() * 6));
+    const destination = Uint8Array.from(
+        { length: LENGTH },
+        () => (Math.random() * 6));
+
+    return Arrow.RecordBatch.new({
+        'lat': Arrow.Float32Vector.from(lat),
+        'lng': Arrow.Float32Vector.from(lng),
+        'origin': Arrow.DictionaryVector.from(values, new Arrow.Int8(), origin),
+        'destination': Arrow.DictionaryVector.from(values, new Arrow.Int8(), destination),
```
Comment (trxcllnt) [inline js/perf/config.ts:46]:
```suggestion
        'origin': Arrow.Vector.new(Arrow.Data.Dictionary(originType, 0, origin.length, 0, null, origin, values)),
        'destination': Arrow.Vector.new(Arrow.Data.Dictionary(destinationType, 0, destination.length, 0, null, destination, values)),
```

**Review by trxcllnt — APPROVED** [review]

**Comment by trxcllnt** [comment]
Does this include (or supersede) https://github.com/apache/arrow/pull/10361?

**Comment by domoritz** [comment]
Ahh, yes, forgot to note. This should be merged after #10361.

**Comment by kou** [comment]
@domoritz Could you rebase on master to merge this?

File: js/src/io/interfaces.ts [hunk 3]
```diff
@@ -174,6 +174,6 @@ export class AsyncQueue<TReadable = Uint8Array, TWritable = TReadable> extends R
         if (this._closedPromiseResolve) {
             return true;
         }
-        throw new Error(`${this} is closed`);
+        throw new Error(`AsyncQueue is closed`);
```
Comment (domoritz) [inline js/src/io/interfaces.ts:177]: Without this, I got an error like "could not stringily non primitive".

**Comment by domoritz** [comment]
@kou done.

**Comment by kou** [comment]
Thanks.
I'll merge this.

**Pull request closed without merging**